src/route.rs

```diff
@@ -138,6 +140,33 @@ impl<'a, State: 'static> Route<'a, State> {
         Ok(())
     }
 
+    /// Serve a index file statically.
+    ///
+    /// The file will be streamed from disk, and a mime type will be determined
+    /// based on magic bytes.
```

Comment:
```suggestion
    /// based on the resolution defined in [Body::from_file](http_types::Body::from_file)
```

although that documentation should be improved as well.  Alternatively, this could explain that it's guessed based on the binary data _or_ the path extension for ascii data, but probably that logic should be documented in just one place.